models/br_bd_execucao_estadual/code/validate_sp.py

```
@@ -0,0 +1,226 @@
+        select safe_cast(ano as int64) as ano,
+               {", ".join(f"sum({BR.format(src)}) as {src}" for src, _ in COLS)}
+        from `{STAGING}.sp_despesa` where {IS_TOTAL} group by ano
+    ), ours as (
+        select ano, {", ".join(f"sum({dst}) as {src}" for src, dst in COLS)}
+        from `{PROD}.despesa_anual` group by ano
+    )
+    select p.ano,
+        {", ".join(f"round(o.{src} - p.{src}, 2) as d_{src}" for src, _ in COLS)},
+        {", ".join(f"p.{src} as p_{src}" for src, _ in COLS)}
+    from published p join ours o using (ano) order by p.ano
+    """
```

Ruff (0.16.2)

[error] Possible SQL injection vector through string-based query construction

(S608)